aws/src/main/java/org/apache/iceberg/aws/AwsClientFactories.java

```diff
@@ -48,9 +49,23 @@ public static AwsClientFactory defaultFactory() {
   }
 
   public static AwsClientFactory from(Map<String, String> properties) {
-    String factoryImpl = PropertyUtil.propertyAsString(
-        properties, AwsProperties.CLIENT_FACTORY, DefaultAwsClientFactory.class.getName());
-    return loadClientFactory(factoryImpl, properties);
+    AwsClientFactory awsClientFactory;
```

Comment:
I read the code a few more times, and I think we can simplify this part. It makes more sense to move this logic to `GlueCatalog` (L113) where we create the client factory. This limits all the changes to GlueCatalog, and it feels more natural to check GLUE_LAKEFORMATION_ENABLED there.